org-code-javabuilder/lib/src/main/java/org/code/javabuilder/LambdaRequestHandler.java

```diff
@@ -113,27 +110,22 @@ public String handleRequest(Map<String, String> lambdaInput, Context context) {
     JavabuilderContext.getInstance().destroyAndReset();
     this.trackStartupPerformance();
 
-    // TODO: Because we reference the logger object throughout the codebase via
-    // Logger.getLogger(MAIN_LOGGER), we need to set it up in the same scope as code execution to
-    // prevent the instance from being garbage collected. Instead, we should store a global
-    // reference to this logger object and access that directly rather than via
-    // Logger.getLogger(MAIN_LOGGER)
```

Comment:
@sanchitmalhotra126 just double-checking -- am i handling this TODO appropriately with my change below?

Reply:
Ah, this is my bad sorry, I definitely should have been more explicit about this task :) we don't actually need (or want to) move any `addHandler/setUseParentHandlers/removeHandler` code into `LoggerUtils` for this task (just yet). For now, we just want to replace any logging methods (e.g. `.info`, `.warning`, `.severe`) with corresponding methods on `LoggerUtils`. So we shouldn't need to change much in LambdaRequestHandler, WebSocketServer, or AWSSystemExitHelper. Changing those logging methods to use `LoggerUtils` will help us minimize the surface area where `Logger.getLogger(MAIN_LOGGER)` is called, so we can then deal with the garbage collection issue in a separate task (details below).

Context: Logger.getLogger returns a logger object from the logging system, but has a quirk that it may be garbage collected at any point "if a strong reference to the Logger is not kept" ([docs](https://docs.oracle.com/javase/7/docs/api/java/util/logging/Logger.html)). So hypothetically, two calls to `Logger.getLogger(MAIN_LOGGER)` may return different objects if the first is garbage collected before the second call. We just _happen_ to get around this today because store a reference to the Logger object returned from Logger.getLogger(MAIN_LOGGER) in this method at line 126. This method (handleRequest) is the entrypoint into the lambda system and wraps all code execution, so by storing that reference, we keep it around for the entire session. Therefore any calls to Logger.getLogger(MAIN_LOGGER) do end up returning the same reference. However, that doesn't feel great :) it just kind of happens to magically work and can be easily broken if the code is moved. So the follow-up here that I have mind, once we've moved all the logging methods to LoggerUtils, is to move the add/remove handler logic into another unit that can also handle storing the strong reference to the logger object.

Reply:
ahh, okay -- that's no problem! easy enough 😄

Reply:
although, now i have a new question -- if we aren't moving any handler helpers to LoggerUtils, i don't think there's any action i can take on [JAVA-680](https://codedotorg.atlassian.net/browse/JAVA-680) because `ExceptionHandlerTest` only uses `Logger.getLogger` to call those handler helpers:

https://github.com/code-dot-org/javabuilder/blob/1a0befa265ae992f810e130bbfefb1ce17470388/org-code-javabuilder/lib/src/test/java/org/code/javabuilder/ExceptionHandlerTest.java#L36-L44

so that test refactor is blocked on creating the unit that stores the strong reference to the logger, right? is that a separate task?

Reply:
Right yeah, so moving Logger.getLogger methods to LoggerUtils was a prereq for deflaking this test, but looks like all usages in `ExceptionHandler` had already been converted prior to this PR :) the fix to deflake the test would just be to use static mocking to mock static methods on `LoggerUtils` (so we're not dealing with the internals of LoggerUtils calling Logger.getLogger), but that can be handled as a separate task.

Reply:
ahhhh, okay, cool -- thank you! since that's the only action for JAVA-680 i'll just open a separate PR for that